packages/flutter_tools/lib/src/version.dart

```diff
@@ -209,10 +209,13 @@ class FlutterVersion {
 
     if (beenAWhileSinceWarningWasPrinted && installationSeemsOutdated && await newerFrameworkVersionAvailable()) {
       printStatus(versionOutOfDateMessage(frameworkAge), emphasis: true);
-      stamp.store(
+      final Future<Null> saveWarningStampFuture = stamp.store(
         newTimeWarningWasPrinted: _clock.now(),
       );
-      await new Future<Null>.delayed(kPauseToLetUserReadTheMessage);
+      await Future.wait<Null>(<Future<Null>>[
+        saveWarningStampFuture,
```

Comment:
feel free to just inline the future-returning calls in the Future.wait call, fwiw.